config/config-api/src/main/java/com/thoughtworks/go/config/BasicCruiseConfig.java

```diff
@@ -335,6 +339,14 @@ private EnvironmentsConfig mergeEnvironmentConfigs() {
             return environments;
         }
 
+        private SCMs mergeScms() {
+            SCMs scms = BasicCruiseConfig.this.scms;
+            for (PartialConfig part : this.parts) {
+                scms.addAll(part.getScms());
```

Comment:
You'll also have to take care of deduplication here. If user specified 2 scms which have same definition, we should make it into 1 at global level.

Reply:
Agree. Will also need to take care of the case where an existing global SCM (in the config) is re-defined in the config repo.

Reply:
Doesn't this already cover it? https://github.com/gocd/gocd/blob/e2b72aa28a5204e3947212b8e0af1686c939f788/server/src/main/java/com/thoughtworks/go/config/update/PartialConfigUpdateCommand.java#L49-L53

Reply:
@ibnc if user specified 2 SCMs which have equal definitions but skipped `scm_id`, then post constructor callback would have assigned 2 random uuids and this would not prevent the duplicated instances from being inserted here.